webviews/src/views/RokuAutomationView/AutoRunsEditor.svelte

```diff
@@ -0,0 +1,395 @@
+<script>
+    import { Trash, Copy, TriangleDown } from 'svelte-codicons';
+    import { dropzone, draggable } from './dnd';
+
+    export let runs;
+    export let selectedRun;
+
+    let runTable;
+    let runNameInput;
+    let runNameDialog;
+    let runNameDialogValue;
+    let confirmDialog;
+    let showContent = false;
+
+    const toggleDropDown = () => {
+        showContent = !showContent;
+    };
+
+    const selectRun = (e) => {
+        const run = getRunFromEvent(e);
+        selectedRun = run;
+    };
+
+    const moveSelection = (e) => {
+        if (["ArrowUp", "ArrowDown"].includes(e.key)) {
+            const index = runs.findIndex((r) => r.name === selectedRun) + (e.key === "ArrowUp" ? -1 : 1);
+            selectedRun = runs[(index + runs.length) % runs.length].name;
+        }
+    };
+
+    const copyRun = (e) => {
+        const run = getRunFromEvent(e);
+        selectedRun = run;
+
+        showRunNameDialog(`Copy of ${run}`)
+            .then((runName) => {
+                const steps =
+                    runs.find((r) => r.name === selectedRun)?.steps ?? [];
+                selectedRun = runName;
+                runs = [
+                    { name: runName, steps: structuredClone(steps) },
+                    ...runs.filter((r) => r.name !== runName)
+                ];
+            })
+            .catch((e) => {
+                // do nothing
+            });
+    };
+
+    const deleteRun = (e) => {
+        const run = getRunFromEvent(e);
+        selectedRun = run;
+        new Promise((resolve, reject) => {
+            let confirmDialogCallback;
+            confirmDialog.addEventListener(
+                'click',
+                (confirmDialogCallback = (e) => {
+                    const button = e?.target?.id;
+                    if (button === 'YES') {
+                        resolve();
+                    } else if (button === 'NO') {
+                        reject();
+                    } else {
+                        return;
+                    }
+                    e.stopPropagation();
+                    confirmDialog.removeEventListener(
+                        'click',
+                        confirmDialogCallback
+                    );
+                    confirmDialog.close();
+                })
+            );
+            confirmDialog.showModal();
+        })
```

Comment:
Let's move most of this to native svelte actions on the confirm dialog itself. You can store the selected run for use in that separate function

Reply:
yep